stl/inc/tuple

```diff
@@ -1079,23 +1079,48 @@ constexpr decltype(auto) apply(_Callable&& _Obj, _Tuple&& _Tpl) noexcept(
         make_index_sequence<tuple_size_v<remove_reference_t<_Tuple>>>{});
 }
 
+#if _HAS_CXX20
 #if _HAS_CXX23
 template <class _Ty, _Tuple_like _Tuple, size_t... _Indices>
 #else // ^^^ _HAS_CXX23 / !_HAS_CXX23 vvv
 template <class _Ty, class _Tuple, size_t... _Indices>
 #endif // ^^^ !_HAS_CXX23 ^^^
-constexpr _Ty _Make_from_tuple_impl(_Tuple&& _Tpl, index_sequence<_Indices...>) noexcept(
-    is_nothrow_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>) {
-    // construct _Ty from the elements of _Tpl
-    static_assert(is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>,
-        "the target type must be constructible from the fields of the argument tuple (N4950 [tuple.apply]/4).");
+    requires is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD declval<_Tuple>()))...>
+constexpr _Ty _Make_from_tuple_impl(_Tuple&& _Tpl, index_sequence<_Indices...>)
+#else // ^^^ _HAS_CXX20 / !_HAS_CXX20 vvv
+template <class _Ty, class _Tuple, size_t... _Indices>
+constexpr _Ty _Make_from_tuple_impl(_Tuple&& _Tpl, index_sequence<_Indices...>,
+    enable_if_t<is_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>, int> = 0)
+#endif // ^^^ !_HAS_CXX20 ^^^
+    noexcept(is_nothrow_constructible_v<_Ty, decltype(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl)))...>) {
     return _Ty(_STD get<_Indices>(_STD forward<_Tuple>(_Tpl))...);
 }
 
+#if _HAS_CXX23
+template <class _Ty, _Tuple_like _Tuple, class _Seq = make_index_sequence<tuple_size_v<remove_reference_t<_Tuple>>>,
+    class = void>
+_INLINE_VAR constexpr bool _Can_make_from_tuple = false;

```

Comment:
These stuffs require C++17, so we should directly use `inline`.

BTW, we can drop `inline` in a future version where MSVC finally gets CWG-2387 implemented, see DevCom-10518531.

Reply:
Done, use `inline` directly now.